pkg/asset/installconfig/gcp/validation.go

```diff
@@ -172,35 +170,80 @@ func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.
 	return allErrs
 }
 
-// ValidatePreExitingPublicDNS ensure no pre-existing DNS record exists in the public
-// DNS zone for cluster's Kubernetes API.
-func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
+// ValidatePreExistingPrivateDNS ensure that the PrivateZone exists in the cluster domain
+func ValidatePreExistingPrivateDNS(client API, ic *types.InstallConfig) error {
+
+	if ic.Platform.GCP.PrivateDNSZone != nil && ic.Platform.GCP.PrivateDNSZone.ID != "" {
```

Comment:
The logic in this validation can be flattened significantly by returning early when `ic.Platform.GCP.PrivateDNSZone == nil or ic.Platform.GCP.PrivateDNSZone.ID == ""` When this field is not specified, there are no validations necessary.

```
if ic.Platform.GCP.PrivateDNSZone != nil && ic.Platform.GCP.PrivateDNSZone.ID != "" {
    return nil
}
project := ic.Platform.GCP.PrivateDNSZone.ProjectID
...
```